lib/view_component/base.rb

```diff
@@ -160,6 +160,12 @@ def request
 
     attr_reader :content, :view_context
 
+    # The parent controller used for previewing components.
+    # Defaults to Rails::ApplicationController. This should be set early
+    # in the initialization process and should be set to a string.
+    mattr_accessor :preview_base_controller
+    @@preview_base_controller = "Rails::ApplicationController"
+
```

Comment:
Maybe we can place this in the previewable module (?) https://github.com/github/view_component/blob/master/lib/view_component/previewable.rb